### PR TITLE
Fix clippy warnings

### PR DESCRIPTION
**Pull request by tesuji**
**Note**: Please squash these commits before merge.

### CODE REVIEW

**Comment by rbtcollins** [comment]
I have  a large branch that will conflict, I think it will be easier for you to rebase and eject a couple of your clippy fixes than the other way around - please hold off on merging a couple days.

**Comment by kinnison** [comment]
@lzutao Could you please rebase, and resubmit?

**Comment by tesuji** [comment]
Sorry for the delay! I pushed new changes.

File: src/config.rs
Comment (kinnison) [inline src/config.rs:98]: Is this a new lint I was previously unaware of?

File: src/diskio/threaded.rs
Comment (kinnison) [inline src/diskio/threaded.rs]: Why two if statements rather than one with the two handler calls in it?

File: src/diskio/threaded.rs
Comment (kinnison) [inline src/diskio/threaded.rs:125]: In fact, move the `let mut prev_files...` above the `if let` above and it can be one `if` with three calls in it perhaps?

File: src/diskio/threaded.rs
Comment (kinnison) [inline src/diskio/threaded.rs:148]: Ditto, why not one `if` statement rather than two?

**Review by kinnison — CHANGES_REQUESTED** [review]
Mostly looks sane.  I only commented on the first `Typename -> Self` conversion, but I'd like to know which lint is making that desirable because to me it's less readable in some cases, though better in others.

A suggestion is present for some of your refactoring, otherwise looks good.

File: src/config.rs
Comment (tesuji) [reply src/config.rs:98]: I just enabled [the lint](https://rust-lang.github.io/rust-clippy/master/index.html#use_self) in this PR.

Comment (kinnison) [reply src/config.rs:98]: Aah I see.  This one is a *pedantic* lint and I'm not sure I completely care for it.  I won't say "no" this time, but it feels a bit overly pedantic for my liking.  Can make it harder to tell what something is if it's overused IMO.

**Comment by tesuji** [comment]
Honestly, I don't know how `handler` and code around that work
so I am afraid of side effect that `handler` could cause. But if you
are OK with this, I change it now.

**Comment by kinnison** [comment]
The changes look good, but the build failed.  And differently this time -- the `dist` code simply stopped by the looks of it.

**Comment by tesuji** [comment]
The `dist` test is passed on my [Azure build](https://dev.azure.com/taolzu/rustup.rs/_build/results?buildId=335&view=logs&jobId=f1715f20-0e3b-5a41-557a-4a8f867c6c58&taskId=fdc67eb1-e0b2-5604-0801-0cd1f81bb2cc&lineStart=1037&lineEnd=1038&colStart=1&colEnd=1).

**Comment by tesuji** [comment]
Oops! You forgot to squash this PR before merging.

**Comment by kinnison** [comment]
> Oops! You forgot to squash this PR before merging.

I didn't really see a need to squash it.  Each commit was clear what it was doing.

**Pull request merged**
